waspc/data/Generator/templates/server/src/types/index.ts

```diff
@@ -5,24 +5,24 @@ import {
   {=# entities =}
   type {= name =},
   {=/ entities =}
- } from "../entities"
+} from "../entities"
 
-export type Query<Entities extends WaspEntity[] = [], Result = unknown> = Operation<Entities, Result>
+export type Query<Entities extends WaspEntity[] = [], Input = unknown, Result = unknown> = Operation<Entities, Input, Result>
 
-export type Action<Entities extends WaspEntity[] = [], Result = unknown> = Operation<Entities, Result>
+export type Action<Entities extends WaspEntity[] = [], Input = unknown, Result = unknown> = Operation<Entities, Input, Result>
 
 {=# isAuthEnabled =}
-export type AuthenticatedQuery<Entities extends WaspEntity[] = [], Result = unknown> = 
-  AuthenticatedOperation<Entities, Result>
+export type AuthenticatedQuery<Entities extends WaspEntity[] = [], Input = unknown, Result = unknown> = 
+  AuthenticatedOperation<Entities, Input, Result>
 
-export type AuthenticatedAction<Entities extends WaspEntity[] = [], Result = unknown> = 
-  AuthenticatedOperation<Entities, Result>
+export type AuthenticatedAction<Entities extends WaspEntity[] = [], Input = unknown, Result = unknown> = 
+  AuthenticatedOperation<Entities, Input, Result>
```

Comment:
Please change the default type of `Input` into `never`. This will allow subtyping when the input's type isn't specified.
### Explanation
Function types are _variant_ for result types, but _contravariant_ for argument types. Variance tells us whether applying a type constructor to a type preservers or flips the subtype/supertype relationship. More precisely (taken from the SO answer linked below):
> Variance has to do with how a generic type `F<T>` varies with respect to its type parameter `T`. If you know that `T extends U`, then variance will tell you whether you can conclude that `F<T> extends F<U>`, conclude that `F<U> extends F<T>`, or neither, or both.

Sources:
 - https://stackoverflow.com/a/66411491
 - https://en.wikipedia.org/wiki/Covariance_and_contravariance_(computer_science)

### Example
Try compiling the following snippets:
```typescript
export const createTask: CreateTask<unknown> = async (task: Pick<Task, 'description'>, context) => {
  // body
}
```
```typescript
export const createTask: CreateTask<never> = async (task: Pick<Task, 'description'>, context) => {
  // body
}
```

Reply:
Good catch, will fix.

Reply:
Can you verify my update?